fizzbuzz/FizzBuzz.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+class FizzBuzz
+{
+    private $rules = [];
+
+    public function __construct()
+    {
+        $this->rules[] = new MultipleOfThree();
+        $this->rules[] = new MultipleOfFive();
+    }
+
+    public function check($number)
+    {
+        $output = '';
+        foreach ($this->rules as $rule) {
+            $output .= $rule->output($number);
+        }
+
+        if (strlen($output)) {
+            return $output . PHP_EOL;
+        }
+        return $number . PHP_EOL;
+    }
+}
+
+abstract class Multiple
```

Comment:
I will use an interface, because this abstract class is poor. https://sourcemaking.com/refactoring/extract-interface